src/enumerable.cr

```diff
@@ -630,6 +630,12 @@ module Enumerable(T)
   # ```
   # [1, 2, 3, 4, 5].reduce(10) { |acc, i| acc + i } # => 25
   # ```
+  #
+  # The initial value can be of any type, for example a collection:
```

Comment:
I'd honestly rather remove this text and just put both examples together:
> Just like the other variant, but you can set the initial value of the accumulator.
>
> ```
> [1, 2, 3, 4, 5].reduce(10) { |acc, i| acc + i } # => 25
> [1, 2, 3].reduce([] of Int32) { |memo, i| memo.unshift(i) } # => [3, 2, 1]
> ```

Reply:
ok